Test convertability of AD-regular Rank*Tensors

lindsayad opened the pull request:
It was revealed by Ferret that we didn't have a proper
assignment operator for RankThreeTensors for different
template parameters. Now we test convertability between
all the different Rank*Tensors in MOOSE

Refs idaholab/moose#14701

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/502559/) on a8291fb wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/15075/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/15075/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format e9feae96fed5e4e4fb115a5f35f35fa492e673a4`

fdkong reviewed (dismissed):
@lindsayad Could you make sure the PR is able to pass `precheck` before you merge?

lindsayad commented on test/tests/kernels/simple_diffusion/simple_diffusion.i:
```diff
@@ -1,8 +1,8 @@
-  dim = 2
-  nx = 10
-  ny = 10
+  dim = 1
+  nx = 2
+  # ny = 2
```
damn it this is what I get for trying to be funny @fdkong

fdkong replied:
Haha

lindsayad committed:
Test convertability of AD-regular Rank*Tensors

It was revealed by Ferret that we didn't have a proper
assignment operator for RankThreeTensors for different
template parameters. Now we test convertability between
all the different Rank*Tensors in MOOSE

Refs idaholab/moose#14701

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/502931/) on fa56e43 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/15075/site)

This comment will be updated on new commits.

lindsayad commented:
@mangerij I didn't actually need to modify ferret at all. I believe this PR should smooth things over

mangerij commented:
Great!
________________________________
From: Alex Lindsay <notifications@github.com>
Sent: Thursday, April 16, 2020 10:15 PM
To: idaholab/moose <moose@noreply.github.com>
Cc: Mangeri, John <john.mangeri@uconn.edu>; Mention <mention@noreply.github.com>
Subject: Re: [idaholab/moose] Test convertability of AD-regular Rank*Tensors (#15075)


*Message sent from a system outside of UConn.*


@mangerij<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmangerij&data=02%7C01%7Cjohn.mangeri%40uconn.edu%7Cfcbbe377e91849d2fd9b08d7e242df4e%7C17f1a87e2a254eaab9df9d439034b080%7C0%7C0%7C637226649152496213&sdata=QeWnVBoKVdYue8WJ8qrRZHHGRo%2F3UN%2FQtnxyd7jyZhc%3D&reserved=0> I didn't actually need to modify ferret at all. I believe this PR should smooth things over

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fidaholab%2Fmoose%2Fpull%2F15075%23issuecomment-614871503&data=02%7C01%7Cjohn.mangeri%40uconn.edu%7Cfcbbe377e91849d2fd9b08d7e242df4e%7C17f1a87e2a254eaab9df9d439034b080%7C0%7C0%7C637226649152506207&sdata=tzx6jRSsjIdZ%2F%2FK1KEHJMh4%2Bbkemob7El3lH3efIq%2Bs%3D&reserved=0>, or unsubscribe<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FABZ65FG4TTTOW6SFLUCF3M3RM5RNDANCNFSM4MJASJTA&data=02%7C01%7Cjohn.mangeri%40uconn.edu%7Cfcbbe377e91849d2fd9b08d7e242df4e%7C17f1a87e2a254eaab9df9d439034b080%7C0%7C0%7C637226649152506207&sdata=XM9AdTgHM7o1BSszd8xSohPHg7pHNRIMaPXUZ%2BXnmVo%3D&reserved=0>.
